### DISCUSSION TITLE
Don't advertise projects in libre trial

### DISCUSSION THREAD

**Discussion by keunes** [WeblateOrg/weblate#6790] [Ideas]
For AntennaPod I've been setting up a trial integration with Weblate, on a testing repository of the project's website. It seems that despite the repo title & description, and us not advertising the Weblate project anywhere, people are starting to translate the project.

The problem is: as this is just a _trial_ setup and I don't want to put work in preserving translations in the process, all [done translations work](https://github.com/keunes/AP-website-copy) will be trashed. I know there's a warning on the project in Weblate (at least for me as a project admin), but people don't read or don't care.

It would be good, therefore, if free Libre trial projects would not be advertised in the public list, so that translators won't do work that might be lost.

**Comment by nijel** [comment]
The trial projects are not advertised, but are listed. Users get a warning when accessing it:

![image](https://user-images.githubusercontent.com/212189/141788646-b3e61452-c053-4a6b-a742-4cd785115351.png)

Maybe it could say something about not preserving strings as well. Suggestions are welcome, the warning is here:

https://github.com/WeblateOrg/weblate/blob/951057fc7adf3222589d6e92f33e5f997b20f0d3/weblate/templates/snippets/project/billing-libre-trial.html#L11

**Reply by keunes** [reply]
> The trial projects are not advertised, but are listed.

For me that's kinda the same. Unless there's something 'extra' Weblate does on top of including Libre projects in the public list?

**Comment by orangesunny** [comment]
I tried to improve the message, feedback is welcome!
@keunes, @comradekingu

https://github.com/WeblateOrg/weblate/pull/6846

**Comment by comradekingu** [comment]
@keunes Overall it is a great benefit that people find projects and start contributing.

Use the [announcements](https://docs.weblate.org/en/latest/admin/announcements.html) and info field, and/or just lock the project when not working on it.
Tried improving the message, but that isn't a stopgap for leaving projects.
If you add "kingu" I can help set it up and sort out the requirements.

**Comment by orangesunny** [comment]
I did the second version of the message improvement, probably final.

A lock, together with a message, is a great way to deal with too eager translators.

Hiding a project is and won’t be available for projects using Libre plan.

**Reply by keunes** [reply]
> Hiding a project is and won’t be available for projects using Libre plan.

The thing is: I'm not yet on a Libre plan. So I don't expect people to be contributing their time and effort (yet).

**Reply by keunes** [reply]
> A lock, together with a message, is a great way to deal with too eager translators.

To properly test my set-up, I _don't_ want locked resources. Resources are locked automatically if something is wrong, and noticing when they occur and how to fix them is an important part of the learning curve.

**Reply by orangesunny** [reply]
> The thing is: I'm not yet on a Libre plan. So I don't expect people to be contributing their time and effort (yet).

Technically you are, but I understand the confusion. We will inform maintainers better.

> To properly test my set-up, I don't want locked resources. Resources are locked automatically if something is wrong, and noticing when they occur and how to fix them is an important part of the learning curve.

Sure it is. Announcement to the translators is usually enough. Locking should be only for the time when you are not testing at the moment. But is not essential.

**Reply by comradekingu** [reply]
I imagine what you want is to de-list the project and then open it back up once ready.
I wonder if it is possible to lock all languages except for developer_en or whatever it is called.

**Comment by orangesunny** [comment]
More progress in https://github.com/WeblateOrg/weblate/pull/6846 if anybody else would like to join the chat.

There are separate messages for translators and admins now.

**Comment by comradekingu** [comment]
Is there a link to the checklist for libre project setup for any given project that is also available to users?

**Reply by nijel** [reply]
No, that it is currently available only on the billing view.

**Comment by nijel** [comment]
Looking again at https://github.com/WeblateOrg/weblate/pull/6855 and https://github.com/WeblateOrg/weblate/pull/6846, I don't think any of these are good. These should be short messages to set the expectations and not a full guidance of what user can do.

For admin:

* We want to avoid surprise that translators start contributing immediately. 
* The immediate flow of translations is something users new to Hosted Weblate do not expect and might cause issues.

For user:

* The translation might not be in a shape to contribute, so it might make sense to wait with a contribution until it is approved.
* In most cases, it does not matter *what* is going to change - the project might get rejected, it might change the license, or discard all the translations because they forgot to specify CLA.

**Reply by comradekingu** [reply]
I agree, and that is my argument now forwarded by you. How my PR differs I don't know,
but it is easy to point out how the original and the other proposal do.

The energy spent arguing more of the same at least did something, a progress of sorts.
I challenge anyone to do less without making it what I did, or argue why it doesn't or wont work.
What is missing, what words are superfluous, or unclear?

Meanwhile, it could have been fixed [a month ago](https://github.com/WeblateOrg/weblate/pull/6855).
How about giving it a month of testing or however long it takes anyone to come up with a feasible alternative?

**Reply by nijel** [reply]
No it could not, I pretty much agree with @keunes there - that didn't improve understanding of the problem. Omitting the reason (trial period) or focusing only on the license is not really an improvement.

**Reply by comradekingu** [reply]
Well, I agree with you that the underlying reason doesn't matter.
"till it is fully set up and accepted" is the important part for the user.
That it is a "trial period" is nondescript, and then "trial period" needs to be explained.
I wonder how you will do it if not the way I did.
CLA is a licensing instrument, you cannot change one without agreeing to what is essentially a new license. How CLAs are allowed for libre projects with gratis hosting is not my call, but explaining them as different from licensing is going to be interesting.

In any event, I think I am correct and instead of having a message that you committed and I thought was wrong from the start, we have no translations and no testing. One wonders what could possibly go wrong to the point of being worse.

I did it in 20 words. The current alternative is 40, up from 28.
That is a UI improvement.

As you lumped both the shorter and longer into
>These should be short messages to set the expectations and not a full guidance of what user can do.

I wonder what you think can be taken away from the 20?

https://github.com/WeblateOrg/weblate/pull/7019/files
